Add Cloud instructions to five guides

ptgott opened the pull request:
See #10637

Database Access FAQ

- Add tabbed instructions where answers differ for Cloud/Self-Hosted

- Style/grammar tweaks

Dynamic Registration

- While this guide is edition agnostic, I have added the tctl.mdx
  partial so Cloud users know to log in to Teleport before running
  tctl commands.

Configuration reference

- The only part that required different instructions for Cloud users
  was the reference configuration for the Proxy Service. I have
  supplied  the values hardcoded in Teleport Cloud.

CLI reference

- Add ScopedBlocks where examples mention proxy.example.com--use a
  Cloud tenant address for Cloud readers.

- Light style/grammar edits

Architecture

- This guide is mostly edition agnostic. After the overview diagram,
  I used a ScopedBlock to make it clear to Cloud users that the Proxy
  Service uses the Teleport Cloud tenant address.

ptgott committed:
Add Cloud instructions to five guides

See #10637

Database Access FAQ

- Add tabbed instructions where answers differ for Cloud/Self-Hosted

- Style/grammar tweaks

Dynamic Registration

- While this guide is edition agnostic, I have added the tctl.mdx
  partial so Cloud users know to log in to Teleport before running
  tctl commands.

Configuration reference

- The only part that required different instructions for Cloud users
  was the reference configuration for the Proxy Service. I have
  supplied  the values hardcoded in Teleport Cloud.

CLI reference

- Add ScopedBlocks where examples mention proxy.example.com--use a
  Cloud tenant address for Cloud readers.

- Light style/grammar edits

Architecture

- This guide is mostly edition agnostic. After the overview diagram,
  I used a ScopedBlock to make it clear to Cloud users that the Proxy
  Service uses the Teleport Cloud tenant address.

r0mant commented on docs/pages/database-access/faq.mdx:
```diff
@@ -3,76 +3,102 @@ title: Database Access FAQ
 
-Teleport Database Access currently supports PostgreSQL, MySQL, MariaDB and MongoDB
-protocols.
+Teleport Database Access currently supports the PostgreSQL, MySQL, MariaDB and
+MongoDB protocols.
```
Since you're here anyway, can you add Microsoft SQL Server and Redis to the list?

ptgott replied:
Added. I've also added Azure Database to the list of Cloud-hosted databases we support.

r0mant commented on docs/pages/database-access/faq.mdx:
```diff
@@ -3,76 +3,102 @@ title: Database Access FAQ
-RDS, Aurora (except for Serverless version which doesn't support IAM auth),
-Redshift, and GCP Cloud SQL are supported. See available [guides](./guides.mdx)
-for all supported configurations.
+RDS, Aurora (except for the serverless version which doesn't support IAM auth),
```
Nit: Aurora Serverless is actually a product name which is why it was capitalized here.

r0mant commented on docs/pages/database-access/faq.mdx:
```diff
@@ -3,76 +3,102 @@ title: Database Access FAQ
+</TabItem>
+<TabItem scope={["cloud"]} label="Teleport Cloud">
+
+In Teleport Cloud, the Proxy Service uses the following public addresses for
```
> public addresses

The table below doesn't actually list any addresses, I would probably rather say "exposes the following ports".

r0mant reviewed (approved):
lgtm with some minor comments

ulysseskan commented on docs/pages/database-access/reference/configuration.mdx:
```diff
@@ -51,6 +54,34 @@ proxy_service:
+  # Address advertised to PostgreSQL clients.
+  postgres_public_addr: "mytenant.teleport.sh:443"
+  # Address advertised to Mongo clients. If not set, public_addr is used.
+  mongo_public_addr: "mytenant.teleport.sh:443"
```
Are there additional ports configured if Microsoft SQL Server or Redis is used?

If only one or two database types are used, are all the ports for all the database types still configured on the Proxy?

ptgott replied:
This is based on the current Proxy Service configuration that's hardcoded for Cloud tenants. It doesn't look like there are Proxy Service settings for SQL Server and Redis.

ulysseskan reviewed (approved):
Looks good to me.  Added one question above.

ptgott committed:
Respond to PR feedback

Also remove mongo_public_addr from the Cloud proxy_service config,
since this was included incorrectly.
